lib/Mojolicious/Plugin/OpenAPI.pm

```diff
@@ -144,28 +140,26 @@ sub _add_routes {
 
 sub _before_render {
   my ($c, $args) = @_;
-  return unless $args->{template};    # This and status is set in the case of 404 and 500
-  return unless my $status = $args->{status} || $c->stash('status');
-  return unless my $self = _self($c);
-
-  if ($status eq '500') {
-    $args->{data} = $self->_renderer->(
-      $c, {errors => [{message => 'Internal server error.', path => '/'}], status => 500}
-    );
-    $args->{status} = $c->stash('status') || $status;
-  }
-  elsif (!$c->stash('openapi.op_path')) {
-    $status = 404;
-    $args->{data}
-      = $self->_renderer->($c, {errors => [{message => 'Not found.', path => '/'}], status => 404});
-    $args->{status} = $c->stash('status') || $status;
-  }
-  elsif ($status eq '404') {
-    $status = 501;
-    $args->{data} = $self->_renderer->(
-      $c, {errors => [{message => 'Not implemented.', path => '/'}], status => 501}
+  my $handler = $args->{handler} || 'openapi';
+
+  # Call _render() for response data
+  return if $handler eq 'openapi' and exists $c->stash->{openapi} or exists $args->{openapi};
+
+  # Fallback to default handler for things like render_to_string()
+  return $args->{handler} = $c->app->renderer->default_handler unless exists $args->{handler};
```

Comment:
I'm not entirely sure if this logic will cover all cases where you want to do `$c->render_to_string()`, but hopefully it will. The idea here is that if you're not explicitly rendering "openapi", then I want to reset "handler" variable from the route, back to the default.

Should be covered by https://github.com/jhthorsen/mojolicious-plugin-openapi/blob/feature/error-reporting/t/autorender.t#L11, but please try it out @marcusramberg.